### ISSUE TITLE
Android Support

### ISSUE THREAD

**Issue by smkplus** [Gaxil/Unity-AlembicToVAT#1]
first of all thanks for great plugin
but It don't work in android
there isn't any texture compression
![image](https://user-images.githubusercontent.com/16706911/83109472-b8143900-a0d6-11ea-9b8e-a57d23556729.png)

![Screenshot_20200528-112236_AlembicToVAT](https://user-images.githubusercontent.com/16706911/83109572-dd08ac00-a0d6-11ea-9a73-b15356d5cdb4.jpg)
![Screenshot_20200528-112738_AlembicToVAT](https://user-images.githubusercontent.com/16706911/83109595-e5f97d80-a0d6-11ea-8c08-b4ba74cd8ac2.jpg)

**Comment by smkplus** [comment]
@neonascent hey thanks I will try

**Comment by smkplus** [comment]
@neonascent 
how you could save it as texture
because Gaxil / Unity-AlembicToVAT  will save as .asset in

I tried to convert it to png but It work strange!
File.WriteAllBytes(finalExportPath+ExportFilename+"_position.png",  positionTexture.EncodeToPNG ());
File.WriteAllBytes(finalExportPath+ExportFilename+"_normal.png",  normalTexture.EncodeToPNG ());

**Comment by alapparikh** [comment]
@smkplus I'm facing the same issue. Encoding it as png gives rise to some colouring artifacts and banding. Did you manage to figure this out?

**Comment by alapparikh** [comment]
@smkplus I'm facing the same issue. Were you able to figure this out eventually? Tried a bunch of different import settings for the PNG but no luck

**Comment by smkplus** [comment]
@alapparikh I tried to convert it to png but I had some problems.
so I made a tool for converting Alembic on Android and it worked for me.
I just set the vertex position data to mesh.
what kind of Alembic you have?

**Comment by alapparikh** [comment]
That's great! I'm not sure what you mean by what kind of Alembic, but it's exported from Oculus Quill so I need to read in the vertex positions and colours and then play it back on Android. The file has a varying number of vertices and eventually needs to run on an Oculus Quest (Android). 

I modified the code in this repository to also write out the colour data per sample and it works, but the textures themselves are pretty large in size so as of now that's my main problem. 

As I understand, you simply read the Alembic file at runtime on Android and set vertex positions? Or is there an intermediate step? If your plugin is open source or you're willing to share it that would be amazing.

**Comment by smkplus** [comment]
> That's great! I'm not sure what you mean by what kind of Alembic, but it's exported from Oculus Quill so I need to read in the vertex positions and colours and then play it back on Android. The file has a varying number of vertices and eventually needs to run on an Oculus Quest (Android).
> 
> I modified the code in this repository to also write out the colour data per sample and it works, but the textures themselves are pretty large in size so as of now that's my main problem.
> 
> As I understand, you simply read the Alembic file at runtime on Android and set vertex positions? Or is there an intermediate step? If your plugin is open source or you're willing to share your approach in some more detail that would be amazing.

Hi I mean what do you want to do?
ok fire alembic. my plugin just works for constant vertex numbers, not for simulations
I just exported array of vertex positions from alembic and I used in unity
this article help you to understand the concept of VAT
https://medium.com/tech-at-wildlife-studios/texture-animation-techniques-1daecb316657

can you pull request what you did? because I want to fix the problem

**Comment by alapparikh** [comment]
Yes sure I can make a pull request. Give me a bit. And thanks for the link.

The code isn't quite complete but you can take a look and see if it's useful to you.

**Comment by alapparikh** [comment]
@smkplus Sorry I haven't had the time to revisit my code, but uploading my script here in case you find it useful. It essentially contains updates to also write out a colour texture asset. The files encoded in png do not work properly. 

[ATV_Editor.zip](https://github.com/Gaxil/Unity-AlembicToVAT/files/5223087/ATV_Editor.zip)

**Comment by smkplus** [comment]
@alapparikh ok thanks I will check it
